Makefile.am

```diff
@@ -1,40 +1,56 @@
 
-bin_SCRIPTS =
+bin_PROGRAMS =
+BUILT_SOURCES =
 CLEANFILES =
 DISTCLEANFILES =
 EXTRA_DIST =
 
 resource_files = $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/speedwagon.gresource.xml)
-speedwagon.gresource: speedwagon.gresource.xml $(resource_files)
-	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir)  $<
-CLEANFILES += speedwagon.gresource
-
-do_subst = sed \
-	-e 's|@bindir[@]|$(bindir)|g' \
-	-e 's|@pkglibdir[@]|$(pkglibdir)|g' \
-	-e 's|@pkgdatadir[@]|$(pkgdatadir)|g' \
-	-e 's|@GJS_CONSOLE[@]|$(GJS_CONSOLE)|g'
-
-src/config.js: src/config.js.in Makefile
-	$(AM_V_GEN) $(MKDIR_P) $(builddir)/src && \
-	$(do_subst) $< > $@
-CLEANFILES += src/config.js
-
-js_sources = \
-	src/config.js \
-	src/eos_speedwagon.js \
+src/speedwagon-resources.h: speedwagon.gresource.xml Makefile $(resource_files)
+	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate-header --c-name eos_speedwagon $<
+src/speedwagon-resources.c: speedwagon.gresource.xml Makefile $(resource_files)
+	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate-source --c-name eos_speedwagon $<
+
+EXTRA_DIST += \
+	$(resource_files) \
+	speedwagon.gresource.xml \
 	$(NULL)
+CLEANFILES += \
+	src/speedwagon-resources.h \
+	src/speedwagon-resources.c \
+	$(NULL)
+BUILT_SOURCES += \
+	src/speedwagon-resources.h \
+	src/speedwagon-resources.c \
+	$(NULL)
+
+src/speedwagon-service.h: Makefile $(top_srcdir)/data/com.endlessm.Speedwagon.xml
+	$(AM_V_GEN) $(MKDIR_P) src && \
+	$(GDBUS_CODEGEN) \
+		--interface-prefix com.endlessm. \
+		--generate-c-code src/speedwagon-service \
+		--c-namespace Eos \
+		$(top_srcdir)/data/com.endlessm.Speedwagon.xml
+src/speedwagon-service.c: src/speedwagon-service.h
+	@: # generated as a side-effect
```

Comment:
Like this, but the recipe (`@:`) is not needed. `make` would simply chain the dependencies and expect that the `.c` file is created by running the `.h` target.